pkg/cpu/cpu_linux.go

```diff
@@ -64,85 +84,69 @@ func processorsGet(ctx *context.Context) []*Processor {
 		curProcAttrs[key] = value
 	}
 
-	// Build a set of physical processor IDs which represent the physical
-	// package of the CPU
-	setPhysicalIDs := make(map[int]bool)
-	for _, attrs := range procAttrs {
-		pid, err := strconv.Atoi(attrs["physical id"])
-		if err != nil {
+	// Iterate on /sys/devices/system/cpu/cpuN, not on /proc/cpuinfo
+	Entries, _ := ioutil.ReadDir(paths.SysDevicesSystemCPU)
+	for _, lcore := range Entries {
+		reg := regexp.MustCompile("^cpu([0-9]+)$")
+		if !reg.MatchString(lcore.Name()) {
 			continue
 		}
-		setPhysicalIDs[pid] = true
-	}
 
-	for pid := range setPhysicalIDs {
-		p := &Processor{
-			ID: pid,
-		}
-		// The indexes into the array of attribute maps for each logical
-		// processor within the physical processor
-		lps := make([]int, 0)
-		for x := range procAttrs {
-			lppid, err := strconv.Atoi(procAttrs[x]["physical id"])
-			if err != nil {
-				continue
-			}
-			if pid == lppid {
-				lps = append(lps, x)
-			}
-		}
-		first := procAttrs[lps[0]]
-		p.Model = first["model name"]
-		p.Vendor = first["vendor_id"]
-		numCores, err := strconv.Atoi(first["cpu cores"])
-		if err != nil {
-			continue
-		}
-		p.NumCores = uint32(numCores)
-		numThreads, err := strconv.Atoi(first["siblings"])
-		if err != nil {
+		lcoreID, error := strconv.Atoi(lcore.Name()[3:]) // get "N" out of "cpuN"
```

Comment:
(very) minor: since we are already using a regexp, we can evaluate if we can use a group to extract the cpuid right away instead of doing in two stages. Not blocking, can be done in a later PR/commit